src/main/java/com/faforever/client/chat/ChatCategoryItemController.java

```diff
@@ -30,39 +34,65 @@
   private final PreferencesService preferencesService;
   private final ContextMenuBuilder contextMenuBuilder;
 
-  public Label chatUserCategoryRoot;
+  public Node root;
+  public Label categoryLabel;
+  public Label visibilityStateLabel;
+
   private ChatUserCategory chatUserCategory;
+  private String channelName;
+  private ObservableMap<String, ObservableList<ChatUserCategory>> channelNameToHiddenCategoriesProperty;
 
-  void setChatUserCategory(@Nullable ChatUserCategory chatUserCategory) {
+  public void setChatUserCategory(ChatUserCategory chatUserCategory, String channelName) {
```

Comment:
I know this is a draft but I don't see how associating the channel with the chatUserCategory would help. I think it would be  better to just have the association at the channeltabcontroller level

Reply:
It was a temporary change. Reverted.